feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,737 @@
+				t.Errorf("ERROR: InDrops counter got %d, want 0", drop.InterfaceBlock.GetOversubscription())
+			}
+		}
+		if drop.LookupBlock != nil {
+			if drop.LookupBlock.GetAclDrops() > 0 {
+				t.Errorf("ERROR: AclDrops counter got %d, want 0", drop.LookupBlock.GetAclDrops())
+			}
+			if drop.LookupBlock.GetForwardingPolicy() > 0 {
+				t.Errorf("ERROR: ForwardingPolicy got %d, want 0", drop.LookupBlock.GetForwardingPolicy())
+			}
+			if drop.LookupBlock.GetFragmentTotalDrops() > 0 {
+				t.Errorf("ERROR: FragmentTotalDrops got %d, want 0", drop.LookupBlock.GetFragmentTotalDrops())
+			}
+			if drop.LookupBlock.GetIncorrectSoftwareState() > 0 {
+				t.Errorf("ERROR: IncorrectSoftwareState counter got %d, want 0", drop.LookupBlock.GetIncorrectSoftwareState())
+			}
+			if drop.LookupBlock.GetInvalidPacket() > 0 {
+				t.Errorf("ERROR: InvalidPacket counter got %d, want 0", drop.LookupBlock.GetInvalidPacket())
+			}
+			if drop.LookupBlock.GetLookupAggregate() > 0 {
+				t.Errorf("ERROR: LookupAggregate counter got %d, want 0", drop.LookupBlock.GetLookupAggregate())
+			}
+			if drop.LookupBlock.GetNoLabel() > 0 {
+				t.Errorf("ERROR: NoLabel counter got %d, want 0", drop.LookupBlock.GetNoLabel())
+			}
+			if drop.LookupBlock.GetNoNexthop() > 0 {
+				t.Errorf("ERROR: NoNexthop counter is %d, not zero", drop.LookupBlock.GetNoNexthop())
+			}
+			if drop.LookupBlock.GetNoRoute() > 0 {
+				t.Errorf("ERROR: NoRoute counter is %d, not zero", drop.LookupBlock.GetNoRoute())
+			}
+			if drop.LookupBlock.GetOversubscription() > 0 {
+				t.Errorf("ERROR: Oversubscription counter got %d, want 0", drop.LookupBlock.GetOversubscription())
+			}
+			if drop.LookupBlock.GetRateLimit() > 0 {
+				t.Errorf("ERROR: RateLimit counter got %d, want 0", drop.LookupBlock.GetRateLimit())
+			}
+		}
+	}
+}
+
+func TestInterfaceStatus(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	interfaces := sortedInterfaces(dut.Ports())
+	t.Logf("Interfaces: %s", interfaces)
+	for _, intf := range interfaces {
+		query := gnmi.OC().Interface(intf).State()
+		root := gnmi.Get(t, dut, query)
+
+		if root.GetOperStatus() != oc.Interface_OperStatus_UP {
+			t.Errorf("ERROR: Oper status is not up on interface %s", intf)
+		}
+		if root.GetAdminStatus() != oc.Interface_AdminStatus_UP {
+			t.Errorf("ERROR: Admin status is not up on interface %s", intf)
+		}
+		if root.Type == oc.IETFInterfaces_InterfaceType_UNSET {
+			t.Errorf("ERROR: Type is not present on interface %s", intf)
+		}
+		if root.Description == nil {
+			t.Errorf("ERROR: Description is not present on interface %s", intf)
+		}
+		if root.GetCounters().OutOctets == nil {
+			t.Errorf("Counter OutOctets is not present on interface %s", intf)
+		}
+		if root.GetCounters().InMulticastPkts == nil {
+			t.Errorf("ERROR: Counter InMulticastPkts is not present on interface %s", intf)
+		}
+		if root.GetCounters().InDiscards == nil {
+			t.Errorf("ERROR: Counter InDiscards is not present on interface %s", intf)
+		} else {
```

Comment:
collapse all of these by using `else if` (e.g. `} else if root.GetCounters().GetInDiscards() > 0 {`), here and below

Reply:
Thanks for your comments. Because the test doesn't compare values now, updated the code as below:
```go
		if root.GetCounters().InDiscards == nil {
			t.Errorf("ERROR: Counter InDiscards is not present")
		} else {
			t.Logf("INFO: Counter InDiscards: %d", root.GetCounters().GetInDiscards())
		}
```